test/e2e/handler/examples_test.go

```diff
@@ -21,20 +21,32 @@ import (
 	"fmt"
 
 	. "github.com/onsi/ginkgo/v2"
+
+	nmstate "github.com/nmstate/kubernetes-nmstate/api/shared"
 )
 
 type exampleSpec struct {
-	name       string
-	fileName   string
-	policyName string
-	ifaceNames []string
+	name         string
+	fileName     string
+	policyName   string
+	ifaceNames   []string
+	cleanupState *nmstate.State
 }
 
 // This suite checks that all examples in our docs can be successfully applied.
 // It only checks the top level API, hence it does not verify that the
 // configuration is indeed applied on nodes. That should be tested by dedicated
 // test suites for each feature.
 var _ = Describe("[user-guide] Examples", func() {
+	cleanDNSDesiredState := nmstate.NewState(`dns-resolver:
+    config:
+      search: []
+      server: []
+interfaces:
+- name: eth1
+  state: absent
+`)
+
```

Comment:
Make sense to have an example with this ?

Reply:
>We may add a separate example for cleaning the DNS setting, but we still need to clean the DNS configuration in the AfterEach, otherwise we hit issues with the outer-most AfterEach, that attempts to restore interfaces - because we can't remove the interface if DNS configuration is bound to it (it currently requires specific interface, either dynamic with auto-dns: false of static with static gateway), so the cleanupState is still required.